ckanext/datastore/logic/action.py

```diff
@@ -598,6 +556,53 @@ def datastore_make_public(context, data_dict):
     db.make_public(context, data_dict)
 
 
+def set_datastore_active_flag(model, data_dict, flag):
+    '''
+    Set appropriate datastore_active flag on CKAN resource.
+
+    Called after creation or deletion of DataStore table.
```

Comment:
let's keep the link to `issue #3245: race condition` in the description to explain why we're doing this direct update of the ckan models ourselves for now.